zaza/openstack/charm_tests/ceph/tests.py

```diff
@@ -967,6 +975,17 @@ def test_003_object_storage_and_secondary_block(self):
         content = primary_object_one.get()['Body'].read().decode('UTF-8')
         self.assertEqual(content, obj_data)
 
+        if self.virtual_hosted_bucket_enabled:
+            # test virtual hosted bucket
+            logging.info('Testing virtual hosted bucket')
+            primary_client.Bucket(container_name).Acl().put(ACL='public-read')
+            primary_client.Object(container_name,obj_name).Acl().put(ACL='public-read')
+            public_hostname = zaza_model.get_application_config(self.primary_rgw_app)["os-public-hostname"]["value"]
+            url = primary_endpoint + "/" + obj_name
+            headers = {'host': container_name + "." + public_hostname}
```

Comment:
Minor style nit: suggest to use f-strings for readability

```
url = f"{primary_endpoint}/{obj_name}"
headers = {'host': f"{container_name}.{public_hostname}"}
```

Reply:
this is changed